src/App.js

```diff
@@ -13,15 +57,80 @@ function App() {
     console.log(genreName);
   }
 
+  function onMovieTileClicked(movieName) {
+    setSelectedMovie(movieName);
+  }
+
+  function onSelectionChange(value) {
+    setCurrentSelection(value);
+  }
+
+  function addMovie() {
+    setIsDialogOpen(true);
+    setDialogTitle("ADD MOVIE");
+    setEditedMovie({});
+  }
+
+  function onCloseDialog() {
+    setIsDialogOpen(false);
+  }
+
+  function editMovie(movie) {
+    setIsDialogOpen(true);
+    setDialogTitle("EDIT MOVIE");
+    setEditedMovie(movie);
+  }
+
+  function deleteMovie(movie) {
+    setIsDialogOpen(true);
+    setDialogTitle("DELETE MOVIE");
+    setEditedMovie();
+  }
+
+  function onSaveMovie(movie) {}
+
   return (
     <div className="App" data-testid="app-component">
-      <Counter initialValue={0} />
+      {/* <Counter initialValue={0} /> */}
       <SearchForm initialQuery="Initial Query" onSearch={onSearch} />
-      <GenreSelect
-        onSelect={onSelect}
-        genres={["ALL", "DOCUMENTARY", "COMEDY", "HORROR", "CRIME"]}
-        selectedGenre="COMEDY"
-      />
+      <button onClick={addMovie}>Add movie</button>
+      {isDialogOpen &&
+        (editedMovie ? (
+          <Dialog
+            title={dialogTitle}
+            children={<MovieForm movie={editedMovie} onSubmit={onSaveMovie} />}
```

Comment:
no need to pass the children prop explicitly

`<Dialog><MovieForm /></Dialog>`